selfdrive/car/tests/test_docs.py

```diff
@@ -27,10 +24,13 @@ def test_generator(self):
     assert generated_cars_md == current_cars_md, "Run selfdrive/car/docs.py to update the compatibility documentation"
 
   def test_docs_diff(self):
-    dump_path = os.path.join(BASEDIR, "selfdrive", "car", "tests", "cars_dump")
-    dump_car_docs(dump_path)
-    print_car_docs_diff(dump_path)
-    os.remove(dump_path)
+    base_path = "docs/CARS.md"
+    new_path = CARS_MD_OUT
+
+    result = subprocess.run(['selfdrive/debug/compare_cars_md.py', base_path, new_path], capture_output=True, text=True)
+    diff_output = result.stdout
+
+    assert diff_output == "", "Differences found in CARS.md:\n" + diff_output
```

Comment:
This just tests that the docs differ doesn't crash silently. What you want to modify is `print_docs_diff.py`

Reply:
To clarify, we have `test_docs.py` which is designed to fail if all new changes in `values.py` files haven't been updated to `CARS.md` with `docs.py`. Once the user does that, `test_docs` shouldn't fail any more, so you should revert any changes to that.

Regardless of if CARS.md is out of date or now, we have a bot that comments the diff nicely formatted: https://github.com/commaai/openpilot/blob/38c0fdac473797d81277a7a8d448f2b663277289/.github/workflows/selfdrive_tests.yaml#L252-L304

An example: https://github.com/commaai/openpilot/pull/32420

The bot is what we want to simplify, that includes the complicated checkout process in the GitHub Action I linked.

Reply:
@sshane Thanks for the clarification. We could update this test to work with the new approach where we no longer rely on `dump_car_docs.py` and instead compare `CARS.md` directly using the `print_car_docs_diff` function.

**Suggest change:**
```python
def test_docs_diff(self):
    new_path = CARS_MD_OUT
    print_car_docs_diff(new_path)
```

https://github.com/commaai/openpilot/blob/38c0fdac473797d81277a7a8d448f2b663277289/selfdrive/car/tests/test_docs.py#L29-L33


**Suggest change:**
```yml
  car_docs_diff:
    name: PR comments
    runs-on: ubuntu-latest
    if: github.event_name == 'pull_request'
    steps:
      - uses: actions/checkout@v4
        with:
          submodules: true
          ref: ${{ github.event.pull_request.base.ref }}
      - run: git lfs pull
      - uses: ./.github/workflows/setup-with-retry
      - name: Save car docs diff
        id: save_diff
        run: |
          scons -j$(nproc)
          output=$(python selfdrive/debug/print_docs_diff.py --path docs/CARS.md)
          output="${output//$'\n'/'%0A'}"
          echo "::set-output name=diff::$output"
```
https://github.com/commaai/openpilot/blob/38c0fdac473797d81277a7a8d448f2b663277289/.github/workflows/selfdrive_tests.yaml#L252-L270